thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/wrapper/AnomalyDetectorWrapper.java

```diff
@@ -143,12 +145,15 @@ public DetectionPipelineResult run() throws Exception {
 
     List<Interval> monitoringWindows = this.getMonitoringWindows();
     List<MergedAnomalyResultDTO> anomalies = new ArrayList<>();
+    TimeSeries predictions = new TimeSeries();
     for (Interval window : monitoringWindows) {
       List<MergedAnomalyResultDTO> anomaliesForOneWindow = new ArrayList<>();
       try {
         LOG.info("[New Pipeline] running detection for config {} metricUrn {}. start time {}, end time {}", config.getId(), metricUrn, window.getStart(), window.getEnd());
         long ts = System.currentTimeMillis();
-        anomaliesForOneWindow = anomalyDetector.runDetection(window, this.metricUrn);
+        DetectionOutput output = anomalyDetector.runDetection(window, this.metricUrn);
+        anomaliesForOneWindow = output.getAnomalies();
+        predictions.merge(output.getPredictions());
```

Comment:
The windows in `monitoringWindows` are overlapped with each other. doing a merge/append may not get the right result. Unless the moving window is moved into the library, then we can simply get the predictions for the whole detection range.

Reply:
Why the windows in monitoringWindows are overlapped?